examples/standalone/lrauv_control/lrauv_control.cc

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Check README for detailed instructions.
+ * Usage:
+ *   $ gz sim -r worlds/lrauv_control_demo.sdf
+ *   $ # ./lrauv_control speed_m_s yaw_rad pitch_rad
+ *   $ ./lrauv_control 0.5 0.78 0.174
+ */
+
+#include <atomic>
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <gz/msgs.hh>
+#include <gz/transport.hh>
+
+
+using namespace gz;
+
+// Helper class for the PID controller for
+// linear speed, pitch and yaw angles.
+class Controller
+{
+  public:
+    // Desired state.
+    double targetSpeed = 0;
+    double targetYawAngle =  0;
+    double targetPitchAngle = 0;
+
+    // Errors
+    double errorSpeed = 0;
+    double errorSpeedIntegral = 0;
+    double errorYawAngle = 0;
+    double errorPitchAngle = 0;
+
+    // States to be tracked and controlled.
+    std::atomic<double> speed;
+    std::atomic<double> yawAngle;
+    std::atomic<double> pitchAngle;
+
+    // PID gains and error limits.
+    // PI for speed.
+    double kSpeed = -30;
+    double kISpeed = -0.5;
+    double errorSpeedIntegralMax = 10;
+
+    // P for yaw and pitch control.
+    double kYawAngle = -0.5;
+    double kPitchAngle = 0.6;
+
+    // Set the target states to be tracked,
+    // i.e. linear speed (m/s), pitch and yaw angles (rad).
+    void SetTargets(double _speed,
+        double _yaw, double _pitch)
+    {
+      if (_speed == 0 &&
+          (_yaw != 0 || _pitch != 0))
+      {
+        std::cout << "Speed needs to be non zero for non zero"
+          " pitch and yaw angles" << std::endl;
+        return;
+      }
+
+      targetSpeed = _speed;
+      targetYawAngle = _yaw;
+      targetPitchAngle = _pitch;
+    }
+
+    // Update the state of the vehicle.
+    void UpdateState(double _speed,
+        double _yaw, double _pitch)
+    {
+      speed = _speed;
+      yawAngle = _yaw;
+      pitchAngle = _pitch;
+
+      errorSpeed = targetSpeed - speed;
+      errorSpeedIntegral =
+        std::min(errorSpeedIntegral + errorSpeed, errorSpeedIntegralMax);
+
+      errorYawAngle = targetYawAngle - yawAngle;
+      errorPitchAngle = targetPitchAngle - pitchAngle;
+
```

Comment:
@adityapande-1995 nit: should we remove the extra blank line?

Reply:
Removed https://github.com/gazebosim/gz-sim/pull/1822/commits/043a82db1cf1a97cb5b8f1d18676010e415c0ece